apps/api/v2/src/modules/auth/guard/oauth/refresh-token.guard.ts

```diff
@@ -0,0 +1,37 @@
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { CanActivate, ExecutionContext, Injectable, UnauthorizedException } from "@nestjs/common";
+
+@Injectable()
+export class RefreshTokenGuard implements CanActivate {
```

Comment:
this guard would need to be changed a little, the refresh-token would be sent in the body of the request, just need to check if it's valid

checking the access-token in headers is doable for added security, basically just need to make sure the access_token in the headers matches the user associated with the refresh token, doesn't matter if the access token is expired

Reply:
The `/refresh` endpoint will need some sort of guard as well, what do you think we should do there?

Reply:
I think the guard is just checking if the refresh token is valid

Reply:
and if it is requested with the right client_secret

Reply:
basically this guard will be tied to the endpoint which should be called fron the customer api, so I think it makes sense they should pass the client_secret as headers